src/Api/Validator.php

```diff
@@ -189,6 +189,13 @@ private function check_boolean(Shape $shape, $value)
 
     private function check_string(Shape $shape, $value)
     {
+        if ($shape['jsonvalue']
+            && !is_resource($value)
+            && (is_object($value) || is_array($value))
```

Comment:
I'm a bit confused with those check, maybe we could just simply skip the check like: 
``` php
if ($shape['jsonvalue']) {
    return;
}
```
I think, since you already have error handling in `RestSerializer` than throws error for those objects that couldn't be JSON serialized, and the encode value would be string anyways if succeed, what's the purpose for verifying the actually object type there?